src/PerfView/PerfViewData.cs

```diff
@@ -4122,36 +4123,54 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
+                    stackSource.AddSample(sample);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWRefCountIncArgs>(args => {
+                    sample.Metric = 1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
 
-                    var ccwRefKindName = "CCW " + operation;
-                    var ccwRefKindIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
-                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefKindIndex, stackIndex);
+                    var ccwRefKindName = "CCW RefCountInc";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
 
-                    var ccwRefCountName = "CCW NewRefCnt " + data.NewRefCount.ToString();
-                    var ccwRefCountIndex = stackSource.Interner.FrameIntern(ccwRefCountName);
-                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefCountIndex, stackIndex);
+                    var ccwRefCount = "CCW NewRefCnt " + args.refCount;
+                    var ccwRefCountIndex = stackSource.Interner.FrameIntern(ccwRefCount.ToString());
+                    stackSource.Interner.CallStackIntern(ccwRefCountIndex, stackIndex);
 
-                    var ccwInstanceName = "CCW Instance 0x" + data.COMInterfacePointer.ToString("x");
-                    var ccwInstanceIndex = stackSource.Interner.FrameIntern(ccwInstanceName);
-                    stackIndex = stackSource.Interner.CallStackIntern(ccwInstanceIndex, stackIndex);
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWRefCountDecArgs>(args =>
+                {
+                    sample.Metric = -1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
 
-                    var ccwTypeName = "CCW Type " + data.NameSpace + "." + data.ClassName;
-                    var ccwTypeIndex = stackSource.Interner.FrameIntern(ccwTypeName);
-                    stackIndex = stackSource.Interner.CallStackIntern(ccwTypeIndex, stackIndex);
+                    var ccwRefKindName = "CCW RefCountDec";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    var ccwRefCount = "CCW NewRefCnt " + args.refCount;
+                    var ccwRefCountIndex = stackSource.Interner.FrameIntern(ccwRefCount.ToString());
+                    stackSource.Interner.CallStackIntern(ccwRefCountIndex, stackIndex);
```

Comment:
Please put the instance identifier (it is passed as objectID), as one of the pseudo-fields (as a hex number).  

For extra credit, add another callback that listens to the TaskCCWCreation event, and remembers a table that maps object ID to Type ID (the TaskCCWCreation has the Type ID).   Then use your ObjectID here to look up the TYpe ID and ALSO include a pseudo-field for that.  (it is useful to group things by type, and  as a next step, we can also turn this type ID into a actual name by looking it up in the PDB (but we can save that for later)).

Reply:
done